cleanupInvalidTags.js

```diff
@@ -0,0 +1,95 @@
+// Retrieve
+var MongoClient = require("mongodb").MongoClient;
+var fs = require("fs");
+var ObjectId = require("mongodb").ObjectID;
+
+// Connect to the db
+// Dev
+// MongoClient.connect("mongodb://x:x@localhost:5555/epic", async function(err, client) {
+// Test
+// MongoClient.connect("mongodb://x:x@localhost:5555/epic", async function(err, client) {
+// Local
+MongoClient.connect("mongodb://localhost/epic", async function(err, client) {
+  if (!err) {
+    console.log("We are connected");
+    const db = client.db("epic");
+
+    let typeData = require(process.cwd() + "/null_type");
+    console.log("typeData:", typeData.length);
+
+    let projectPhaseData = require(process.cwd() + "/null_projectPhase");
+    console.log("projectPhaseData:", projectPhaseData.length);
+
+    let authorData = require(process.cwd() + "/null_author");
+    console.log("authorData:", authorData.length);
+
+    let milestoneData = require(process.cwd() + "/null_milestone");
+    console.log("milestoneData:", milestoneData.length);
+
+    for (let z = 0; z < typeData.length; z++) {
+        let object_id = typeData[z]._id.substring(9,33);
+        updateType(db, ObjectId(object_id));
```

Comment:
If the update functions are returning promises but we don't wait for them to resolve anywhere, is it possible that `client.close()` could be called before all of the update functions have resolved? It most likely won't happen, but I think a race condition might exist here.

Reply:
I believe we have resolves after each of the updates (line 62,72,82,92). Then we call client.close() after all get updated. But let me know if you see any error in my understanding!

Reply:
That is right; however, the value being returned from the update functions is actually the promise itself and not the resolved value. In order to avoid the race condition they would need to be waited for somewhere before the connection is closed.

Reply:
I see! I added the `await` operator when calling the update functions. Can you take a look? :-)

Reply:
Just updated promise handling now. thanks for the great suggestions @KitArmstrong